wagtail/snippets/templates/wagtailsnippets/snippets/create.html

```diff
@@ -3,9 +3,9 @@
 {% block titletag %}{% blocktrans trimmed with snippet_type_name=model_opts.verbose_name %}New  {{ snippet_type_name }}{% endblocktrans %}{% endblock %}
 {% block content %}
     {% trans "New" as new_str %}
-    {% include "wagtailadmin/shared/header_with_locale_selector.html" with title=new_str subtitle=model_opts.verbose_name icon="snippet" tabbed=1 merged=1 %}
+    {% include "wagtailadmin/shared/header_with_locale_selector.html" with title=new_str subtitle=model_opts.verbose_name icon="snippet" tabbed=1 merged=1 action_url=None %}
 
-    <form action="{% url 'wagtailsnippets:add' model_opts.app_label model_opts.model_name %}{% if locale %}?locale={{ locale.language_code }}{% endif %}" method="POST" novalidate{% if form.is_multipart %} enctype="multipart/form-data"{% endif %}>
+    <form action="{{ action_url }}" method="POST" novalidate{% if form.is_multipart %} enctype="multipart/form-data"{% endif %}>
```

Comment:
Refactoring this to use `action_url` that gets its value from `get_add_url()`, see:

https://github.com/wagtail/wagtail/blob/1d73aa2cec521f1670aaa42f07c1bdb33ce9b467/wagtail/admin/views/generic/models.py#L157-L161

However, after adding this, the `action_url` in the `wagtailadmin/shared/header_with_locale_selector.html` must be unset so that the Add button does not appear on the top right corner. Other views that extend `CreateView` usually do this by using `{% include ... with ... only %}`, but we still want to pass other context variables to the `include` (i.e. the locale stuff), so I do `action_url=None` for now. Another option would be to explicitly include the locale in `with` and use `only`.

Reply:
I'd prefer explicitly including this rather than excluding I think. It's less likely to have clashes if any more cases like this are added in the future.